### PR TITLE
scraper: Improve handling of ETags in http class and tweak verified beacon logic

### PR DESCRIPTION
**Pull request by jamescowens**
This improves the parsing of the http header for ETags to deal with several defects in the original version and also tweaks the beacon verification logic to allow verification of beacons on the pending list even if they are already active.

### CODE REVIEW

File: src/scraper/scraper.cpp [hunk 1]
```diff
@@ -2230,57 +2230,69 @@ bool ProcessProjectRacFileByCPID(const std::string& project, const fs::path& fil
...
-                for (const auto& entry : GlobalVerifiedBeaconsCopy.mVerifiedMap)
-                {
-                    if (entry.second.cpid == cpid) ++already_verified;
-                }
+                if (entry.second.cpid == cpid) already_verified = true;
 
-                if (!already_verified)
-                {
-                    // Attempt to verify.
+                break;
```
Comment (cyrossignol) [inline src/scraper/scraper.cpp]: This needs to go in the scope of the `if` condition above, right?

File: src/scraper/scraper.cpp [hunk 2]
```diff
@@ -2230,57 +2230,69 @@ bool ProcessProjectRacFileByCPID(const std::string& project, const fs::path& fil
...
-                    IncomingVerifiedBeacons.timestamp = GetAdjustedTime();
+                        IncomingVerifiedBeacons.timestamp = GetAdjustedTime();
 
-                    // We do NOT want to add a just verified CPID to the statistics this iteration,
-                    // because we may be halfway through processing the set of projects. Instead, add to the
-                    // incoming verification map (above), which will be handled in the calling function once
-                    // all of the projects are gone through. This will become a verified beacon the next time
-                    // around.
-                    continue;
+                        just_verified = true;
```
Comment (cyrossignol) [inline src/scraper/scraper.cpp]: Do we need to check `just_verified` somewhere? It looks like it's unused.

Comment (jamescowens) [reply src/scraper/scraper.cpp]: Actually we don't, according to my truth table.

File: src/scraper/http.cpp [hunk 3]
```diff
@@ -211,29 +212,35 @@ std::string Http::GetEtag(
...
+        for (const auto& elem : header_line_elements)
         {
-            continue;
-        }
+            std::string output = elem;
 
-        std::string header_name = line.substr(0, 4);
-        boost::to_lower(header_name);
+            boost::algorithm::trim(output);
+            boost::to_lower(output);
```
[the remark this answers is not in the document]
Comment (jamescowens) [reply src/scraper/http.cpp]: Yeah. I should fix it.

File: src/scraper/scraper.cpp (hunk 2 above)
Comment (jamescowens) [reply src/scraper/scraper.cpp]: Removed the flag and simplified the truth table.

**Review by cyrossignol — APPROVED** [review]

**Pull request merged**